### PR TITLE
libnpupnp: add

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
did not expect that...

**Comment by neheb** [comment]
No idea how to proceed.

File: subprojects/packagefiles/libnpupnp/meson.build [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+if get_option('tools')
+  npupnp_sources += [
+    'src/api/upnptools.cpp',
+  ]
+endif
+
+libnpupnp = library('npupnp',
+  npupnp_sources,
+  include_directories : libnpupnp_incdir,
+  dependencies : [ winsock_dep, iphlpapi_dep, nsl_dep, socket_dep, threads_dep, libmicrohttpd_dep, libcurl_dep, expat_dep ],
```
Comment (eli-schwartz) [inline subprojects/packagefiles/libnpupnp/meson.build:86]: This also has a library version upstream which should be retained, `-version-info 6:1:2` in libtool terms is  `version: '4.2.1'` in meson terms (and produces `libnpupnp.so.4.2.1`).

File: subprojects/packagefiles/libnpupnp/meson.build [hunk 2]
```diff
@@ -0,0 +1,215 @@
+project('npupnp', 'cpp',
```
Comment (eli-schwartz) [inline subprojects/packagefiles/libnpupnp/meson.build:1]: Hmm, shouldn't this be libnpupnp? That is what is in AC_INIT().

**Review by eli-schwartz — COMMENTED** [review]
The libmicrohttpd thing on Windows could be solved either by adding libmicrohttpd to the WrapDB too, or disabling an optional component if not found... but libmicrohttpd doesn't seem to be optional.

**Comment by jpakkane** [comment]
Since the deps are not available as wraps, you need to disable building on Windows in the json files at the root.

**Comment by jpakkane** [comment]
It's a bit unexpected, but you need to set `linux_only` to `true` rather than adding the name to `skip_windows`.

**Comment by neheb** [comment]
Set,

In other news, I tried to make a meson.build file for libmicrohttp but it seems a lot more work than this...

**Comment by eli-schwartz** [comment]
I kinda... really dislike the idea of using linux_only, which is *supposed* to indicate that the source code is written for POSIX and thus isn't expected to build with MSVC, for "don't build on Windows because missing dependencies".

At the very least, this needs an explanatory comment for why on earth the software doesn't build on Windows even though it's supposed to.

(I would much prefer a libmicrohttpd port obviously.)

**Comment by neheb** [comment]
Added comment.

**Comment by jpakkane** [comment]
⚔️ conflicts.

**Comment by neheb** [comment]
Joy.

**Comment by neheb** [comment]
Fixed conflicts.

**Comment by jpakkane** [comment]
Thanks.

**Pull request merged**